### ISSUE TITLE
Attacks with special effects in FOW not fully hidden

### ISSUE THREAD

**Issue by LynxAbraxas** [civctp2/civctp2#299]
It seems attacks with special effects (like conversion, franchising, stealing advances) if happening in FOW (e.g. done by the AI) are not fully hidden: The map focuses on the attack position and the sound of the special attack is played, giving away the attack to human players, and can become an annoyance if happening many times between turns.

**Comment by MartinGuehmann** [comment]
Actually, I like it when the sounds are played, this way you know that something is going on. But of course it shouldn't focus there.

**Comment by LynxAbraxas** [comment]
> this way you know that something is going on

True, but that is a kind of "cheating" since it is some info only humans can make use of, i.e. the AI so far cannot, and being in FOW should be covered to be concise with the game concept, otherwise one could argue that even the animation should be shown even in FOW. And it can become annoying when you have shower of special attack sounds each turn.

**Comment by LynxAbraxas** [comment]
I can also observe special effects for units that are invisible to me like conversion, franchising, stealing advances. This gives away the unit in a way that the AI cannot make use of, so in a way it is also an unfair advantage to human player.

**Comment by williamd67** [comment]
My thoughts on this topic.

Currently the End-turn behavior is influenced by two settings:
1. center map
2. show enemy moves
* If both are disabled sounds of the special effects are played and in case the special effect is on screen an animation; for me that is hard to follow and not very informative;
* In case center maps is enabled, and show enemy moves is disabled; the map jumps to the place were the special effect is happening, and before the animation or the sound is finished, jumps to the next special effect; for me this is also hard to follow and in most cases before I can interpret the effect there is a jump to another place;
* In case both center maps and show enemy moves is enabled: I can, in most cases, follow the actions and effects; this has a negative side-effect that it takes quiet some time to see all enemy moves, especially with larger maps and further in the game (can be sped up by making the animations shorter but still);

So next to @LynxAbraxas remark that it is unfair to the AI-player, we could, in my opinion, improve the End-turn process for the human player as well, by:
* only play animations and sounds that involve the human player so no animations and sounds for the actions between AI-players (or other human-players if applicable)
* or, make it clear (slow enough) for the human-player so (s)he can both understand and enjoy the special effects; although this will slow down his/her process.

Do you have any other ideas, suggestions, or improvements?

**Comment by LynxAbraxas** [comment]
> - In case center maps is enabled, and show enemy moves is disabled; the map jumps to the place were the special effect is happening, and before the animation or the sound is finished, jumps to the next special effect; for me this is also hard to follow and in most cases before I can interpret the effect there is a jump to another place;

True, for me too and it can happen that multiple animations overlap, e.g. a franchise followed by expelling the corporate branch.
However, I really had to disable "show enemy moves" after building globe sat because it would otherwise take far too long to get to the new turn.

> - only play animations and sounds that involve the human player so no animations and sounds for the actions between AI-players (or other human-players if applicable)

This, I think, I like best, but involvement of the human player then should also include just watching a scene, e.g. a special attack between two AI-players within the vision of a  spy from a human player, which could be quite a challenge to realize.

**Comment by LynxAbraxas** [comment]
@williamd67 Isn't this solved by one of your PRs that are already merged? If so, which one was it?

**Comment by LynxAbraxas** [comment]
@williamd67 Isn't this solved by one of your PRs that are already merged? If so, which one was it?

**Comment by MartinGuehmann** [comment]
> I can also observe special effects for units that are invisible to me like conversion, franchising, stealing advances. This gives away the unit in a way that the AI cannot make use of, so in a way it is also an unfair advantage to human player.

Well, the AI knows where everything is on the map, it just prioritizes the stuff it can see. So I don't see here an unfair advantage for the human player.

**Comment by williamd67** [comment]
@LynxAbraxas No I do not think that this is solved by any of my submissions. There were some improvements targeting hidden units, yet that was not effecting the 'end-turn' behavior.

**Comment by LynxAbraxas** [comment]
> > I can also observe special effects for units that are invisible to me like conversion, franchising, stealing advances. This gives away the unit in a way that the AI cannot make use of, so in a way it is also an unfair advantage to human player.
> 
> Well, the AI knows where everything is on the map, it just prioritizes the stuff it can see. So I don't see here an unfair advantage for the human player.

True, the AI knows all but ignores some of that knowledge, i.e. does not make use of it.
The problem I see here is that a human has a hard time to ignore such knowledge;-) so I'd still vote for not closing this issue in regard to @williamd67 answer, as it can be an advantage for a human player and despite that it takes away a bit of the intended surprise of FOW.
